In [16]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [17]:
df = pd.read_csv(r'C:\Users\ashmi\OneDrive\Desktop\Data science project\Data_Science_Project\Dataset\goldstock.csv')
df.head()

Unnamed: 0        Date   Close    Volume    Open    High     Low
0           0  2024-01-19  2029.3  166078.0  2027.4  2041.9  2022.2
1           1  2024-01-18  2021.6  167013.0  2009.1  2025.6  2007.7
2           2  2024-01-17  2006.5  245194.0  2031.7  2036.1  2004.6
3           3  2024-01-16  2030.2  277995.0  2053.4  2062.8  2027.6
4           4  2024-01-12  2051.6  250946.0  2033.2  2067.3  2033.1

In [24]:
# Drop the unessary column Unnamed
df.drop(columns=['Unnamed: 0'],inplace=True)

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2511 entries, 0 to 2510
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    2511 non-null   object 
 1   Close   2511 non-null   float64
 2   Volume  2511 non-null   float64
 3   Open    2511 non-null   float64
 4   High    2511 non-null   float64
 5   Low     2511 non-null   float64
dtypes: float64(5), object(1)
memory usage: 117.8+ KB


* There are total 2511 rows and 7 columns.
* Each column has no missing values.

In [27]:
# Date formatting to ensure the datetime for proper time series analysis
df['Date'] = pd.to_datetime(df['Date'])

In [29]:
# Sort by Date- Earliest to latest
df = df.sort_values(by='Date',ascending=True)

In [30]:
df.describe()

Date        Close         Volume         Open  \
count                           2511  2511.000000    2511.000000  2511.000000   
mean   2019-01-17 07:28:27.526881792  1498.726085  185970.770609  1498.725528   
min              2014-01-22 00:00:00  1049.600000       1.000000  1051.500000   
25%              2016-07-19 12:00:00  1249.850000  126693.500000  1249.500000   
50%              2019-01-17 00:00:00  1332.800000  175421.000000  1334.000000   
75%              2021-07-17 12:00:00  1805.850000  234832.000000  1805.600000   
max              2024-01-19 00:00:00  2093.100000  787217.000000  2094.400000   
std                              NaN   298.824811   97600.769382   299.118187   

              High          Low  
count  2511.000000  2511.000000  
mean   1508.451454  1488.869932  
min    1062.700000  1045.400000  
25%    1257.300000  1242.350000  
50%    1342.400000  1326.600000  
75%    1815.450000  1793.050000  
max    2098.200000  2074.600000  
std     301.262244   296.417703

* The Dataset minimum Date Start 2014-01-22 to max 2024-01-19.


### Feature Engineering
Consider creating additional features that could impact gold prices, such as:
* Price Change: Calculate the difference between 'Close' and 'Open' prices.
* Price Range: Calculate the difference between 'High' and 'Low' prices.

In [ ]:
# Doing some feature engineering to create addtional features.
